models/migrations/v105.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"crypto/md5"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+
+	"xorm.io/xorm"
+)
+
+func renameExistingUserAvatarName(x *xorm.Engine) error {
+	sess := x.NewSession()
+	defer sess.Close()
+
+	type User struct {
+		ID        int64  `xorm:"pk autoincr"`
+		LowerName string `xorm:"UNIQUE NOT NULL"`
+		Avatar    string
+	}
+	deleteList := make(map[string]struct{})
+	start := 0
+	for {
+		if err := sess.Begin(); err != nil {
+			return fmt.Errorf("session.Begin: %v", err)
+		}
+		users := make([]*User, 0, 50)
+		if err := sess.Table("user").Asc("id").Limit(50, start).Find(&users); err != nil {
+			return fmt.Errorf("select users from id [%d]: %v", start, err)
+		}
+		if len(users) == 0 {
+			break
```

Comment:
sess.Rollback()

Reply:
@lunny, we don't have anything in session here. Is `sess.Find()` included in session ?

Another thing, as `sess.Close` does the `Rollback()` part if anything remains in session, do we still need to call `sess.Rollback()` ?
https://github.com/go-gitea/gitea/blob/aa7c34cf86081b71f796f6e5da60e62a3cda43ce/vendor/github.com/go-xorm/xorm/session.go#L98-L113

Reply:
@masudur-rahman for the first question, I think it's OK.

But this will create serval transactions but not one transaction. So you have to Rollback manually on the `for loop`.

Reply:
Okay.
So, @lunny, do we need to use `session.Rollback()` before returning each error ? or just the section you mentioned ?

Reply:
@masudur-rahman Yes. We could just call `session.Close()` or `session.Rollback()`. Both will rollback if you haven't called `session.Commit`.